src/StructuredLogger/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -456,12 +505,24 @@ private void Write(TargetSkippedEventArgs e)
             WriteDeduplicatedString(e.TargetFile);
             WriteDeduplicatedString(e.TargetName);
             WriteDeduplicatedString(e.ParentTarget);
-            WriteDeduplicatedString(""); // Condition
-            WriteDeduplicatedString(""); // EvaluatedCondition
-            Write(true); // OriginallySucceeded
+            WriteDeduplicatedString(e.Condition); // Condition

```

Comment:
wow, looks like our copy of the Writer was indeed a bit behind! I sure hope nobody was using it for anything important!